bugsnag-plugin-android-anr/src/main/java/com/bugsnag/android/AnrPlugin.kt

```diff
@@ -18,6 +18,13 @@ internal class AnrPlugin : Plugin {
             // The only check that will work across all Android versions is the isNativeMethod call.
             return javaTrace.first().isNativeMethod
         }
+
+        private const val stackUnwindOverhead = 5
```

Comment:
This raises a few questions for me:

- Is the number of frames consistent for all versions of Android?
- Would it make sense to consider trimming frames for handled errors from the NDK too?
- How will this affect the grouping of errors on the dashboard, as it will affect the top frame of the stacktrace?

If this has been considered then this changeset looks fine to me (once the other comment has been resolved)

Reply:
I think we need to revisit this in general. The original requirements included trimming the top of the stack trace for ANRs, and in fact the entire purpose of this particular work order was to bring it in line with those requirements.

But in general we now have an issue with what should be present in what stack traces:

- Should NDK traces be trimmed the same as for ANRs?
- Should Java traces be trimmed in the same manner when the android project is part of React or Unity? (basically the same problem, one level higher).
- Maybe all traces should have the top trimmed regardless, so that the top is the actual culprit rather than our trace gathering code?

We should definitely have consistency one way or the other with this.

In this particular case, however, we have direct control over the calls leading to the stack trace, so it will always be 5 unless we change our trace gathering code in the NDK plugin.

Reply:
Send me link deposits